appstore/validator.go

```diff
@@ -121,6 +123,26 @@ func (c *Client) Verify(req IAPRequest, result interface{}) error {
 	defer resp.Body.Close()
 
 	err = json.NewDecoder(resp.Body).Decode(result)
+	if err != nil {
+		return err
+	}
+
+	// Always verify your receipt first with the production URL; proceed to verify with the sandbox URL if you receive
+	// a 21007 status code
+	//
+	// https://developer.apple.com/library/content/technotes/tn2413/_index.html#//apple_ref/doc/uid/DTS40016228-CH1-RECEIPTURL
+	r, ok := result.(*IAPResponse)
```

Comment:
This code makes the developers use ``IAPResponse`` struct.
But AppStore also offers iOS6 style receipt as the response.
In that case, developer can't this method.
I fixed the argument as ``interface{}`` in #29 for the same reason.

Reply:
@jun06t That makes sense. The type assertion `r, ok := result.(*IAPResponse)` will take appropriate action if and only if the result is of type IAPReponse. If it is any other type, it will gracefully return nil. 

For IAPReponse, it will automatically attempt to hit the sandbox if status from production is 21007. Any other types like iOS6 style receipt will just proceed normally as before. This code will still preserve the ability for the developer to explicitly specify sandbox. 

In our production environment where this code is being run, we have many users using production receipts but our team of devs use sandbox receipts to work on other parts of our system.

Reply:
How about defining a new type, so that it can work for both ios6/ios7~ version receipts? The only thing you care here is the status code, right.
HttpStatus struct {
   Status             int                  `json:"status"`
}

Reply:
I like the idea of removing the IAP_ENVIRONMENT. I originally kept it as it allows the developer to go straight to sandbox if they wanted to do so. I'm a bit torn between which way would be the best, but I'm going to remove it just to see how much complexity it will remove (since there is some smelly code to keep this functionality intact). 

I saw your branch in your fork and love the idea with the struct. I'll put that in here

Reply:
I am glad to hear you liked the idea. I'll check the code again once this PR is updated.